ui-v2/app/routes/dc/acls.js

```diff
@@ -25,7 +25,11 @@ export default Route.extend(WithBlockingActions, {
                 // take the user to the legacy acls
                 // otherwise just refresh the page
                 if (get(item, 'token.AccessorID') === null) {
-                  return this.transitionTo('dc.acls');
+                  // returning false for a feedback action means even though
+                  // its successful, please skip this notification and don't display it
+                  return this.transitionTo('dc.acls').then(function() {
+                    return false;
+                  });
```

Comment:
This false means 'don't show the successful notification'. All my problems would be solved here if I did an extra API request to `/v1/acls/list` before the redirect to see if I get a 403 or not. My concern with doing that is, that I then immediately redirect to the listing page with requests the exact same endpoint again. I'd rather not do that.

Reply:
What level of this catches the `false` the feedback service?

Reply:
I'm not sure it's relevant to what your doing, but the Transition class has a `followRedirects` which settles on the final redirection - we use that in vault so that flash messages don't disappear when there's multiple redirects. https://emberjs.com/api/ember/3.4/classes/Transition/methods?anchor=followRedirects

Reply:
So I think if `followRedirects` works for you that'd be a better approach. Returning booleans feels like a fragile contract (ember's own actions work this way for action bubbling...and it's not my favorite - closure actions are much nicer IMO).

Reply:
Just to answer you q on the which level catches the `false`. It's the end of the promise chain, so goes all the way back here:

https://github.com/hashicorp/consul/blob/5a0a6e18f9208cce4a23af9a358e8a9c86934a95/ui-v2/app/services/feedback.js#L27

.. and I check for the `=== false` here:

https://github.com/hashicorp/consul/blob/5a0a6e18f9208cce4a23af9a358e8a9c86934a95/ui-v2/app/services/feedback.js#L30

I did give `followRedirects` a go, but I couldn't get it to solve this particular problem, I think the multiple redirects are part of the problem but not all of it ... maybe..?

Defo a good one to know about though.